### DISCUSSION TITLE
run script with delimiter oracle / syntax

### DISCUSSION THREAD

**Discussion by travelliu** [dbeaver/dbeaver#16448] [Q&A] [answered]
#### System information: 
- Operating system: macOS 12.3.1 x86_64
- DBeaver version:  22.0.4 
- Additional extensions

#### Connection specification:

![2022-05-07 at 00 13](https://user-images.githubusercontent.com/7792329/167171646-e8187692-41fa-4415-82ee-9e4fc8ef3bf4.png)

#### Describe the problem you're observing:
![2022-05-07 at 00 13](https://user-images.githubusercontent.com/7792329/167171712-d137349d-f24e-4162-ac5f-62ff574c47ef.png)

![2022-05-07 at 00 12](https://user-images.githubusercontent.com/7792329/167171562-2d98c7f4-d952-40fe-bfc0-1ce8b60933b8.png)

#### Steps to reproduce, if exist:

```sql
CREATE VIEW a1 AS SELECT * FROM dual
/
CREATE VIEW a2 AS SELECT * FROM dual
/
```

**Comment by E1izabeth** [comment, marked as answer]
Hi, @travelliu! DBeaver uses `;` as a statement delimiter by default, but you can redefine it in settings:

![image](https://user-images.githubusercontent.com/28875055/167701201-44b8793f-8827-4045-8b4e-f17cc806bfe5.png)

Does it resolve the issue?

**Comment by travelliu** [comment]
tks.
```
CREATE VIEW a11 AS SELECT * FROM dual;
/
CREATE VIEW a12 AS SELECT * FROM dual;
/

CREATE OR REPLACE PACKAGE ORA_MTK.TTTTTT1
AS
-- Package header
FUNCTION testfunc4(var NUMBER) return number;
FUNCTION testfunc5(var NUMBER) return number;
END TTTTTT1;
/

CREATE OR REPLACE PACKAGE BODY ORA_MTK.TTTTTT1
AS
-- Package body
FUNCTION testfunc4(var NUMBER) return NUMBER
IS
BEGIN
    RETURN var;
END testfunc4;
END TTTTTT1;
/
```
Run the above script normally after modification

**Comment by travelliu** [comment]
@E1izabeth   Can I change it to the default for Oracle?
https://github.com/dbeaver/dbeaver/blob/devel/plugins/org.jkiss.dbeaver.ext.oracle/src/org/jkiss/dbeaver/ext/oracle/model/OracleSQLDialect.java

![2022-05-11 at 14 35](https://user-images.githubusercontent.com/7792329/167784049-e65ff4c4-9865-4967-a322-2e8c60cecbb3.png)

change to return new String[]{";", "/"};

**Reply by E1izabeth** [reply]
I think it can be done.

**Reply by travelliu** [reply]
@E1izabeth  modify statement delimiter ; to /

run 

![2022-05-31 at 22 02](https://user-images.githubusercontent.com/7792329/171192310-bb07e563-c1a4-4b1c-8f88-c6f8ec614362.png)

JDBC trace log 

```sql 
=======================================================
SELECT 1
```

How to support delimiter / syntax. Does not affect division operations ？

delimiter  Syntax /. commonly used in create statements. Such as

```sql
create table xxxx ()
...
/
create view xxx
...
/

create function xxx
...
/
create package xx
...
/
```

**Comment by travelliu** [comment]
@E1izabeth ?

**Reply by E1izabeth** [reply]
I think there is no way to prevent conflict with division for now

**Reply by travelliu** [reply]
OK.
